lib/singleton-client/pseudo.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+# Copyright 2022 VMware, Inc.
+# SPDX-License-Identifier: EPL-2.0
+
+module Sgtn
+  module Pseudo # :nodoc:
+    protected
+
+    def pickup_locale(locale, _component)
+      locale == PSEUDO_LOCALE ? PSEUDO_LOCALE : super
```

Comment:
if locale=pseudo, will return pseudo translation?
I don't see how to trigger and get the pseudo from Singleton service, the mocked testing just show get from a file but not from a pseudo locale. Could you explain a bit how remote pseudo work?

Reply:
> if locale=pseudo, will return pseudo translation?
Yes. It will

It is limited by current client design, client will get latest source and add pseudo tags on source strings. This should meet your fundamental pseudo requirements. As we've discussed before, it's hard to get pseudo from server to reflect source collection status in current design. I've tried that way and the logic was complex and some issues didn't find a solution. We can discuss more on this offline.

Reply:
i think it has to update the cache by another request to remote latest with pseudo locale, you can consider to implement this by another PR.